app/build.gradle

```diff
@@ -64,4 +66,10 @@ dependencies {
     // Navigation component
     implementation "androidx.navigation:navigation-fragment-ktx:$nav_version"
     implementation "androidx.navigation:navigation-ui-ktx:$nav_version"
+    // Dagger Hilt
+    implementation("com.google.dagger:hilt-android:2.44")
+    kapt("com.google.dagger:hilt-android-compiler:2.44")
+}
+kapt {
+    correctErrorTypes = true
 }
```

Comment:
Can you please explain why you added this

Reply:
it allow references to generated code and tells the Kapt tool to attempt to automatically correct certain error types during annotation processing